compiler-core/src/error.rs

```diff
@@ -127,6 +127,18 @@ pub enum Error {
         reason: InvalidProjectNameReason,
     },
 
+    #[error("{module} is not a valid module name")]
+    InvalidModuleName { module: String },
+
+    #[error("{module} is not module")]
+    ModuleDoesNotExist { module: String },
+
+    #[error("{module} does not have a main function")]
+    ModuleDoesNotHaveMainFunction { module: String },
+
+    #[error("{module} has the wrong arity so it can not be run.")]
+    MainFunctionHasWrongArity { module: String, arity: usize },
```

Comment:
```suggestion
    InvalidModuleName { module: SmolStr },

    #[error("{module} is not module")]
    ModuleDoesNotExist { module: SmolStr },

    #[error("{module} does not have a main function")]
    ModuleDoesNotHaveMainFunction { module: SmolStr },

    #[error("{module} has the wrong arity so it can not be run.")]
    MainFunctionHasWrongArity { module: SmolStr, arity: usize },
```

We can keep it as a SmolStr to remove the code and extra work converting to the mutable string type.

Reply:
Done. `InvalidModuleName` is still a string though because we never converted to `SmolStr` for that one.